src/components/form-control/form-control.js

```diff
@@ -0,0 +1,97 @@
+/*
+MIT License
+
+Copyright (c) 2018 Uber Technologies, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+// @flow
+import * as React from 'react';
+import {getOverride, getOverrideProps} from '../../helpers/overrides';
+import {STYLETRON_PROP_MAPPER} from './constants';
+import {
+  Label as StyledLabel,
+  Caption as StyledCaption,
+  ControlContainer as StyledControlContainer,
+} from './styled-components';
+import type {FormControlPropsT} from './types';
+
+function mapStyletronProps(props: {}, mapper: {}) {
+  return Object.keys(props).reduce((newProps, propName) => {
+    const newName = mapper[propName] && `$${propName}`;
+    if (newName) {
+      newProps[newName] = props[propName];
+    }
+    return newProps;
+  }, {});
+}
+
+export default class FormControl extends React.Component<FormControlPropsT> {
+  static defaultProps = {
+    overrides: {},
+    label: null,
+    caption: null,
+    error: false,
+  };
+
+  render() {
+    const {
+      overrides: {Label: LabelOverride, Caption: CaptionOverride},
+      label,
+      caption,
+      error,
+      children,
+    } = this.props;
+
+    const onlyChildProps = React.Children.only(children).props;
+    const sharedProps = mapStyletronProps(
+      onlyChildProps,
+      STYLETRON_PROP_MAPPER,
+    );
+    sharedProps.$error = this.props.error || sharedProps.$error;
```

Comment:
It seems like this solution turned out pretty elegant. 👍

Reply:
I suppose the discussion in #131 is actually sort of relevant to this. 

There may be cases where someone may want to add a wrapper element between their FormControl and Input element, or if someone wants two inputs side-by-side under the same label (credit card + cvv code, etc). React context doesn't help us here, but maybe for those cases maybe we still want to support optioanlly passing `disabled`, `size`, `required`, etc, to the top-level form-control?